server/services/color_utils.py

```
@@ -0,0 +1,112 @@
+    if words and words[-1] in BASE_COLORS:
+        return [words[-1]]
+
+    # Fallback: return the original lowercased
+    return [lower]
+
+
+def hex_to_hsl(hex_color: str) -> tuple[float, float, float]:
+    """Convert hex color to HSL. Returns (h, s, l) where h is 0-360, s and l are 0-1."""
+    hex_color = hex_color.lstrip("#")
+    r, g, b = (int(hex_color[i:i+2], 16) / 255.0 for i in (0, 2, 4))
+    h, l, s = colorsys.rgb_to_hls(r, g, b)
```

Ruff (0.15.5)

[error] Ambiguous variable name: `l`

(E741)